cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AbstractNotificationSubscriberService.java

```diff
@@ -60,36 +58,28 @@
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
-import java.util.Collection;
 import java.util.Map;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
-import javax.annotation.Nullable;
 
 /**
- * Subscribe to notification TMS topic and update schedules in schedule store and job queue
+ * Abstract class that fetches notifications from TMS
  */
-class NotificationSubscriberService extends AbstractIdleService {
-  private static final Logger LOG = LoggerFactory.getLogger(NotificationSubscriberService.class);
+public abstract class AbstractNotificationSubscriberService extends AbstractIdleService {
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractNotificationSubscriberService.class);
   // Sampling log only log once per 10000
   private static final Logger SAMPLING_LOG = Loggers.sampling(LOG, LogSamplers.limitRate(10000));
   private static final Gson GSON = new Gson();
 
+  private final CConfiguration cConf;
   private final Transactional transactional;
-  private final MultiThreadMessagingContext messagingContext;
   private final DatasetFramework datasetFramework;
+  private final MultiThreadMessagingContext messagingContext;
   private final MultiThreadDatasetCache multiThreadDatasetCache;
-  private final CConfiguration cConf;
-  private ExecutorService taskExecutorService;
   private volatile boolean stopping = false;
 
-
   @Inject
-  NotificationSubscriberService(MessagingService messagingService,
-                                CConfiguration cConf,
-                                DatasetFramework datasetFramework,
-                                TransactionSystemClient txClient) {
+  protected AbstractNotificationSubscriberService(MessagingService messagingService, CConfiguration cConf,
+                                                  DatasetFramework datasetFramework, TransactionSystemClient txClient) {
     this.cConf = cConf;
     this.messagingContext = new MultiThreadMessagingContext(messagingService);
```

Comment:
@maochf and I talked offline - agreed that we can use this as long as the fetcher is not used inside a transaction itself.